src/test/java/mate/academy/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,216 @@
+                objectMapper.readValue(result.getResponse().getContentAsString(),
+                CategoryResponseDto.class);
+        EqualsBuilder.reflectionEquals(expected, actual);
+    }
+
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Get a category by id")
+    void getCategoryById_ValidCategoryId_ShouldReturnCategoryResponseDto() throws Exception {
+        MvcResult result = mockMvc.perform(get("/categories/{id}", TEST_CATEGORY_ID)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        CategoryResponseDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), CategoryResponseDto.class);
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(TEST_CATEGORY_ID, actual.getId());
+    }
+
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Get books list by category id")
+    @Sql(scripts = "classpath:database/categories_books/add-books-to-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/categories_books"
+                + "/delete-data-from-categories-books-table.sql",
+                executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void getBooksByCategory_ValidCategory_ShouldReturnListBookDto() throws Exception {
+        MvcResult result = mockMvc.perform(get("/categories/{id}/books", TEST_CATEGORY_ID)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDtoWithoutCategoryIds[] actual = objectMapper.readValue(
+            result.getResponse().getContentAsString(), BookDtoWithoutCategoryIds[].class);
+
+        Assertions.assertEquals(TEST_CATEGORY_BOOKS_QUANTITY, actual.length);
+        Assertions.assertTrue(Arrays.stream(actual)
+                .anyMatch(bookDtoWithoutCategoryIds -> bookDtoWithoutCategoryIds.getIsbn().equals(
+                "978-8996989274")));
+        Assertions.assertTrue(Arrays.stream(actual)
+                .anyMatch(bookDtoWithoutCategoryIds -> bookDtoWithoutCategoryIds.getTitle().equals(
+                "verdict")));
+    }
+
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Find all categories")
+    void findAll_GiveAllCategories_ShouldReturnAllCategories() throws Exception {
+        MvcResult result = mockMvc.perform(get("/categories")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        JsonNode rootNode = objectMapper.readTree(result.getResponse().getContentAsString());
+        JsonNode contentNode = rootNode.get("content");
+        CategoryResponseDto[] actual = objectMapper.treeToValue(contentNode,
+            CategoryResponseDto[].class);
+
+        Assertions.assertEquals(TEST_CATEGORY_AMOUNT, actual.length);
```

Comment:
looks a bit complex, will this work?
```suggestion
Page<CategoryResponseDto> page = objectMapper.readValue(
    result.getResponse().getContentAsString(),
    new TypeReference<Page<CategoryResponseDto>>() {}
);
Assertions.assertEquals(TEST_CATEGORY_AMOUNT, page.getContent().size());
```